sway/input/cursor.c

```diff
@@ -20,9 +22,35 @@ static void cursor_update_position(struct sway_cursor *cursor) {
 	cursor->y = y;
 }
 
+static struct wlr_surface *layer_surface_at(struct sway_output *output,
+		struct wl_list *layer, double ox, double oy, double *sx, double *sy) {
+	struct sway_layer_surface *sway_layer;
+	wl_list_for_each_reverse(sway_layer, layer, link) {
+		struct wlr_surface *wlr_surface =
+			sway_layer->layer_surface->surface;
+		double _sx = ox - sway_layer->geo.x;
+		double _sy = oy - sway_layer->geo.y;
+		struct wlr_box box = {
+			.x = sway_layer->geo.x,
+			.y = sway_layer->geo.y,
+			.width = wlr_surface->current->width,
+			.height = wlr_surface->current->height,
+		};
+		// TODO: Test popups/subsurfaces
+		if (wlr_box_contains_point(&box, ox, oy) &&
+				pixman_region32_contains_point(
+					&wlr_surface->current->input, _sx, _sy, NULL)) {
```

Comment:
We'll probably want to wrap this into a wlr function

Reply:
meh

Reply:
alright you convinced me